packages/require/lib/typescript.ts

```diff
@@ -0,0 +1,171 @@
+import path from "path";
+
+import type { CreateOptions, Service } from "ts-node";
+import TruffleConfig from "@truffle/config";
+import { readFileSync } from "fs";
```

Comment:
> Maybe we should make these reads (and the whole compilation pipeline) asynchronous?

I think the effort involved in doing that would be far far greater than the value gained _in this specific context_.

That said, I'd be _very_ strongly in support of efforts to make the entire CLI execution flow async, but I don't think it's within the scope of the problem that I'm currently trying to solve.

Reply:
The commit that you linked changes the return type of the `file` function from `any` to `Promise<any>`, which is a breaking change. Resolving that breaking change is where the bulk of the effort would be on this, as it will force the caller to create an asynchronous context.

Reply:
Ah, I see! Thanks for the explanation. I hadn't realized this existed before. I should have noticed the original implementation was deleted and rewritten in a new location (`git mv` would have helped here, but I should have caught it regardless).

You reasoning definitely makes sense now, since this exported file already (mis)behaves this way. Thanks for clarifying!

Note: if it is as easy as fixing up any uses of `Require.file` (doubted it's this easy) it might not be too complex after all as those contexts seem to be async already.

I'm marking this as Resolved. Thanks!

Reply:
> That said, I'd be very strongly in support of efforts to make the entire CLI execution flow async, but I don't think it's within the scope of the problem that I'm currently trying to solve

This should probably be an issue